traffic_ops/traffic_ops_golang/servicecategory/servicecategories.go

```diff
@@ -187,3 +192,313 @@ WHERE name=$2 RETURNING last_updated`
+		return
+	}
+
+	requestedName := inf.Params["name"]
+	// check if the entity was already updated
+	userErr, sysErr, errCode = api.CheckIfUnModifiedByName(r.Header, inf.Tx, requestedName, "service_category")
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+
+	//update name of a service category
+	query := `UPDATE service_category sc SET
+		name = $1
+	WHERE sc.name = $2
+	RETURNING sc.name, sc.last_updated`
+
+	err := tx.QueryRow(query, sc.Name, requestedName).Scan(&sc.Name, &sc.LastUpdated)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			api.HandleErr(w, r, tx, http.StatusNotFound, fmt.Errorf("service category with name: %s not found", requestedName), nil)
+			return
+		}
+		usrErr, sysErr, code := api.ParseDBError(err)
+		api.HandleErr(w, r, tx, code, usrErr, sysErr)
+		return
+	}
+	alerts := tc.CreateAlerts(tc.SuccessLevel, "service category was updated")
+	api.WriteAlertsObj(w, r, http.StatusOK, alerts, sc)
+	return
+}
+
+// DeleteServiceCategory [Version : V5] function deletes the service category passed.
+func DeleteServiceCategory(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	tx := inf.Tx.Tx
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	name := inf.Params["name"]
+	exists, err := dbhelpers.ServiceCategoryExists(tx, name)
+	if err != nil {
+		api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, err)
+		return
+	}
+	if !exists {
+		api.HandleErr(w, r, tx, http.StatusNotFound, fmt.Errorf("no service category exists for name: %s", name), nil)
+		return
+
+	}
+
+	assignedDeliveryService := 0
+	if err := inf.Tx.Get(&assignedDeliveryService, "SELECT count(service_category) FROM deliveryservice d WHERE d.service_category=$1", name); err != nil {
+		api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, fmt.Errorf("service category delete, counting assigned Delivery Service(s): %w", err))
+		return
+	} else if assignedDeliveryService != 0 {
+		api.HandleErr(w, r, tx, http.StatusBadRequest, fmt.Errorf("can not delete a service category with %d assigned Delivery Service(s)", assignedDeliveryService), nil)
+		return
+	}
+
+	res, err := tx.Exec("DELETE FROM service_category AS sc WHERE sc.name=$1", name)
+	if err != nil {
+		api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, err)
+		return
+	}
+	rowsAffected, err := res.RowsAffected()
+	if err != nil {
+		api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, fmt.Errorf("determining rows affected for delete service_category: %w", err))
+		return
+	}
+	if rowsAffected == 0 {
+		api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, fmt.Errorf("no rows deleted for service_category"))
+		return
+	}
+	alerts := tc.CreateAlerts(tc.SuccessLevel, "service_category was deleted.")
+	api.WriteAlertsObj(w, r, http.StatusOK, alerts, name)
```

Comment:
This response will be only the name of the Service Category, like
```json
{
  "alerts": [{
    "level": "success",
    "text": "service_category was deleted."
  }],
  "response": "test"
}
```
but it *should* be the deleted Service Category like
```json
{
  "alerts": [{
    "level": "success",
    "text": "service_category was deleted."
  }],
  "response": {
    "name": "test",
    "lastUpdated": "05-05-2023T00:00:00Z"
  }
}
```

Reply:
api.WriteAlerts had to be used instead of api.WriteAlertsObj for a delete function as there is no need to return name or last_updated. This is in parallel with V4 delete function as well.